activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -746,10 +747,20 @@ def pk_and_sequence_for(table)
         end
       end
 
-      # Returns just a table's primary key
-      def primary_key(table)
-        pk_and_sequence = pk_and_sequence_for(table)
-        pk_and_sequence && pk_and_sequence.first
+      def primary_keys(table_name) # :nodoc:
+        return nil unless table_name.present?
```

Comment:
Why only this implementation returns `nil`?
I think it is better plural method (e.g. `primary_keys`, `tables`, `indexes` etc) returning `[]` (empty array). If they return `[]`, user can always write `primary_keys.each`.